### ISSUE TITLE
RedirectionalCoordinator.anyCoordinator crash  ‘Pushing a navigation controller is not supported’.

### ISSUE THREAD

**Issue by d2vydov** [quickbirdstudios/XCoordinator#97]
that code works fine:

```
let coordinator = AboutCoordinator(superCoordinator: self)
return .push(coordinator
```

but if you take anyCoordinator, from RedirectionalCoordinator,

```
let coordinator = AboutCoordinator(superCoordinator: self)
return .push(coordinator.anyCoordinator)
```

it will be runtime crashed _‘Pushing a navigation controller is not supported’_

**Comment by d2vydov** [comment]
u can see demo at https://github.com/d2vydov/XCoordinator/commits/feature/fix-redirectional-coordinator

**Comment by d2vydov** [comment]
I tried to debug that, and find the problem, in method `.push` in NavigationTransition 

![image](https://user-images.githubusercontent.com/9073024/58559879-15ac4600-8235-11e9-8cb1-b34c0d798f4e.png)

as you can see, it will be push UINavigationController on UINavigationController

**Comment by pauljohanneskraft** [comment]
Thanks for catching that bug/problem - It results from the AnyCoordinator using the default implementation of the Coordinator protocol for the `viewController` property, which is returning its rootViewController. The rootViewController of a RedirectionCoordinator is the rootViewController of its parent, so what your second code example would do is push the rootViewController of the parent coordinator onto itself...

**Comment by d2vydov** [comment]
@pauljohanneskraft can you help me with fix that?

**Comment by pauljohanneskraft** [comment]
It will require a fix of the AnyCoordinator, which should also have a stored closure for the viewController property.

**Comment by d2vydov** [comment]
Any progress with this issue?

**Comment by d2vydov** [comment]
fixed in #98
